qiskit/providers/models/backendproperties.py

```diff
@@ -122,8 +120,143 @@ def __init__(self, backend_name, backend_version, last_update_date,
         self.backend_name = backend_name
         self.backend_version = backend_version
         self.last_update_date = last_update_date
-        self.qubits = qubits
-        self.gates = gates
         self.general = general
 
+        self._qubits = defaultdict(dict)
+        for qubit, props in enumerate(qubits):
+            formatted_props = {}
+            for prop in props:
+                value = self._apply_prefix(prop.value, prop.unit)
+                formatted_props[prop.name] = (value, prop.date)
+                self._qubits[qubit] = formatted_props
+
+        self._gates = defaultdict(dict)
+        for gate in gates:
+            qubits = _to_tuple(gate.qubits)
+            formatted_props = {}
+            formatted_props['name'] = gate.name
+            for param in gate.parameters:
+                value = self._apply_prefix(param.value, param.unit)
+                formatted_props[param.name] = (value, param.date)
+            self._gates[gate.gate][qubits] = formatted_props
+
         super().__init__(**kwargs)
+
+    def gate_error(self, operation: str, qubits: Union[int, Iterable[int]]):
+        """
+        Return gate error estimates from backend properties.
+        Args:
+            operation (str): The operation for which to get the error.
+            qubits (Union[int, Iterable[int]]): The specific qubits for the operation.
+
+        Raises:
+	        PulseError: If error is True and the property is not found.
+        """
+        try:
+            result = self._gates.get(operation).get(_to_tuple(qubits)).get('gate_error')[0]
+            # Throw away datetime at index 1
+            # return self.gates[operation][_to_tuple(qubits)]['gate_error'][0]
+        except (KeyError, TypeError):
+            raise PulseError("Could not find the desired property.")
+        return result
+
+    def gate_length(self, operation: str, qubits: Union[int, Iterable[int]]):
+        """
+        Return the duration of the gate in units of seconds.
+        Args:
+            operation (str): The operation for which to get the duration.
+            qubits (Union[int, Iterable[int]]): The specific qubits for the operation.
+
+        Raises:
+	        PulseError: If error is True and the property is not found.
+        """
+        try:
+            result = self._gates.get(operation).get(_to_tuple(qubits)).get('gate_length')[0]
+            # Throw away datetime at index 1
+            # return self.gates[operation][_to_tuple(qubits)]['gate_length'][0]
+        except KeyError:
+            raise PulseError("Could not find the desired property.")
+        return result
+
+    def get_gate_property(self,
+                          gate: str = None,
+                          qubits: Union[int, Iterable[int]] = None,
+                          gate_property: str = None) -> Tuple[Any, datetime.datetime]:
```

Comment:
maybe this should be `name`, like below, mostly because it's shorter. I'm not set on it